diff --git a/src/controller.js b/src/controller.js
index 8a48c90..b07cea7 100644
--- a/src/controller.js
+++ b/src/controller.js
@@ -62,7 +62,7 @@ Controller.prototype.setView = function(locationHash) {
 */
 Controller.prototype.showAll = function() {
   var that = this
-  that.model.read(function(data) {
+  that.modal.read(function(data) {
     that.view.render('showEntries', data)
   })
 }
@@ -238,7 +238,10 @@ Controller.prototype._filter = function(force) {
   // re-create the todo item elements, calling:
   //   this.show[All|Active|Completed]();
   if (force || this._lastActiveRoute !== 'All' || this._lastActiveRoute !== activeRoute) {
-    this['show' + activeRoute]()
+    try {
+      this['show' + activeRoute]()
+    }
+    catch (err) { }
   }
 
   this._lastActiveRoute = activeRoute
