diff --git a/examples/vanilla-es6/src/template.js b/examples/vanilla-es6/src/template.js
index 947303b..c0bc34c 100644
--- a/examples/vanilla-es6/src/template.js
+++ b/examples/vanilla-es6/src/template.js
@@ -18,7 +18,7 @@ export default class Template {
 	 */
 	itemList(items) {
 		return items.reduce((a, item) => a + `
-<li data-id="${item.id}"${item.completed ? ' class="completed"' : ''}>
+<li data-id="${item.id}"${!item.completed ? ' class="completed"' : ''}>
 	<div class="view">
 		<input class="toggle" type="checkbox" ${item.completed ? 'checked' : ''}>
 		<label>${escapeForHTML(item.title)}</label>
