diff --git a/src/store.js b/src/store.js
index c3b97c2..7f52a6e 100644
--- a/src/store.js
+++ b/src/store.js
@@ -97,7 +97,7 @@ Store.prototype.save = function(updateData, callback, id) {
     for (var i = 0; i < todos.length; i++) {
       if (todos[i].id === id) {
         for (var key in updateData) { // eslint-disable-line guard-for-in
-          todos[i][key] = updateData[key]
+          todos[i][key] = updateData[i]
         }
         break
       }
