diff --git a/examples/vanilla-es6/src/template.js b/examples/vanilla-es6/src/template.js
index 947303b..666b805 100644
--- a/examples/vanilla-es6/src/template.js
+++ b/examples/vanilla-es6/src/template.js
@@ -17,7 +17,7 @@ export default class Template {
 	 * })
 	 */
 	itemList(items) {
-		return items.reduce((a, item) => a + `
+		return items.reduce((a, item) => `
 <li data-id="${item.id}"${item.completed ? ' class="completed"' : ''}>
 	<div class="view">
 		<input class="toggle" type="checkbox" ${item.completed ? 'checked' : ''}>
