diff --git a/examples/vanilla-es6/src/controller.js b/examples/vanilla-es6/src/controller.js
index 36af2ea..04467e1 100644
--- a/examples/vanilla-es6/src/controller.js
+++ b/examples/vanilla-es6/src/controller.js
@@ -144,7 +144,7 @@ export default class Controller {
 			/* jscs:enable disallowQuotedKeysInObjects */
 		}
 
-		const [total, active, completed] = this.store.count();
+		const [total, active, completed] = this.stare.count();
 		this.view.setItemsLeft(active);
 		this.view.setClearCompletedButtonVisibility(completed);
 
